libs/cdk/src/lib/utils/pipes/async-or-sync.pipe.ts

```diff
@@ -0,0 +1,135 @@
+import { ChangeDetectorRef, EventEmitter, OnDestroy, Pipe, PipeTransform, inject } from '@angular/core';
```

Comment:
Can't we reuse `async` pipe here? Inject the pipe and invoke their transform function only in case of the non-static values. We would not need to support everything on our end

Reply:
That might bring some memory leaks, so I would keep it like that for now.

Reply:
I have another idea. Instead of loading pipe into pipe, I'll create `makeAsync` pipe which will transform any non-awaitable/subscribable variable into subscribable one, and then apply async pipe.

Reply:
That way developer will need to use two pipes on one place

Reply:
Yea, OR he can make it async in his service/component. Since I'm against modifying passed initial data, pipe I think this is the best option here. Calling asyncPipe.transform directly may have unintentional side effects.